filewatcher/filewatcher.go

```diff
@@ -102,22 +124,50 @@ func (r *Result) watcherLoop(
 	softLimit, hardLimit int64,
 	logger log.Wrapper,
 	pollingInterval time.Duration,
+	parseDelay time.Duration,
 ) {
 	forceReload := func(mtime time.Time) {
-		f, err := limitopen.OpenWithLimit(path, softLimit, hardLimit)
+		var f io.ReadCloser
```

Comment:
you can change this part to make `dummyReadCloser` no longer need to implement closer as that's not really used:
```go
var reader io.Reader
...
if isDir {
  reader = dummyReader{
    Path: path, // actually Path field no longer need to be exported either.
  }
} else {
  f, err := limitopen.OpenWithLimit(...)
  if err != nil {
    ...
  }
  defer f.Close()
  reader = f
}
// now you can use reader instead of f
```